src/main/java/com/WebController/ApplicationController.java

```diff
@@ -0,0 +1,54 @@
+package com.WebController;
+
+import com.Model.Applicant;
+import com.Repository.ApplicantRepository;
+import com.Service.MessageSender;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.web.bind.annotation.*;
+import java.io.FileNotFoundException;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.Optional;
+
+@RestController
+public class ApplicationController {
+
+    @Autowired
+    private ApplicantRepository applicantRepository;
+
+    @Autowired
+    private MessageSender messageSender;
+
+    @PostMapping (value = "/postapplication")
+    public Applicant postApplication (@RequestBody Applicant applicant) throws FileAlreadyExistsException {
+        //check if any applicantion already existed
+        if (this.applicantRepository.findAllByApplicantEmailAndDescription(applicant.getApplicantEmail(), applicant.getDescription()).size()==0){
+            return this.applicantRepository.save(applicant);
+        }
+        throw new FileAlreadyExistsException("File exists");
+    }
+
+    @GetMapping (value = "/admin/getapplicants")
+    public Iterable<Applicant> getAllApplicants (@RequestParam("page") int page){
+        return this.applicantRepository.findAll(PageRequest.of(page-1, 10));
+    }
+
+    @PostMapping (value = "/admin/approveapplicant")
```

Comment:
is this endpoint protected and limited to Admin access?

Reply:
Yes, in the security configuration class, both "/admin" and "/admin/**" requires ADMIN_ROLE to get access